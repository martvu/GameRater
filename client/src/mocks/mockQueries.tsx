import {
  ADD_FAVORITES,
  CREATE_REVIEW,
  REMOVE_FAVORITES,
  SIGN_IN_OR_CREATE_USER,
} from '@/lib/mutations';
import {
  GET_GAMES,
  GET_GAME,
  GET_FILTERS,
  GET_GAME_PLATFORMS,
  GET_SEARCH_SUGGESTIONS,
} from '@/lib/queries';

const getGamesMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: undefined,
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesMockedKeyword = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesSignedInMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '1',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            platforms: [1, 2],
            genres: [3, 4],
          },
          // Add more game objects as needed
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getFiltersMock = {
  request: {
    query: GET_FILTERS,
    variables: {},
  },
  result: {
    data: {
      getGenres: [
        { id: 1, name: 'Genre 1' },
        { id: 2, name: 'Genre 2' },
      ],
      getPlatforms: [
        { id: 1, name: 'Platform 1' },
        { id: 3, name: 'Platform 3' },
        { id: 4, name: 'PC (Microsoft Windows)' },
        { id: 2, name: 'Nintendo Switch' },
        { id: 5, name: 'PlayStation 5' },
        { id: 6, name: 'Xbox Series X|S' },
      ],
    },
  },
};

const getGamePlatformsMock = {
  request: {
    query: GET_GAME_PLATFORMS,
    variables: { id: '1' },
  },
  result: {
    data: {
      getGame: {
        name: 'Example Game',
        platforms: [
          {
            id: 3,
            name: 'Platform 3',
          },
          {
            id: 4,
            name: 'PC (Microsoft Windows)',
          },
        ],
      },
    },
  },
};

const signInMock = {
  request: {
    query: SIGN_IN_OR_CREATE_USER,
    variables: {
      userInput: {
        username: 'testuser',
      },
    },
  },
  result: {
    data: {
      signInOrCreateUser: {
        _id: '1',
        username: 'testuser',
        favorites: [],
        reviews: [],
      },
    },
  },
};

const getGameMock = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: '',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};

const getGameMock2 = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: 'testUser',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};
const addFavoritesMock = {
  request: {
    query: ADD_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      addFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [{ _id: '1' }],
      },
    },
  },
};

const removeFavoritesMock = {
  request: {
    query: REMOVE_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      removeFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [],
      },
    },
  },
};

const createReviewMock = {
  request: {
    query: CREATE_REVIEW,
    variables: {
      reviewInput: {
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        user: 'testUser',
        gameID: '1',
      },
    },
  },
  result: {
    data: {
      createReview: {
        _id: 'review1',
        user: 'testUser',
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        gameID: '1',
      },
    },
  },
};

const searchbarSuggestionsMock = {
  request: {
    query: GET_SEARCH_SUGGESTIONS,
    variables: {
      query: 'example',
    },
  },
  result: {
    data: {
      getSearchSuggestions: [
        {
          _id: '1',
          name: 'Example Game',
          cover_image_id: 'cover1',
        },
        {
          _id: '2',
          name: 'Example Game 2',
          cover_image_id: 'cover2',
        },
      ],
    },
  },
};

const allMocks = [
  createReviewMock,
  getGamePlatformsMock,
  getGamesMockedKeyword,
  addFavoritesMock,
  removeFavoritesMock,
  getGamesMock,
  getFiltersMock,
  signInMock,
  getGamesSignedInMock,
  getGameMock,
  getGameMock2,
  searchbarSuggestionsMock,
  // Add more mocks as needed for other queries
];

export {
  allMocks,
  getGamesMock,
  getGamesSignedInMock,
  getFiltersMock,
  signInMock,
};
