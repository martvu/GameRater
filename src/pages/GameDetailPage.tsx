import withLayout from '@/lib/withLayout';
import Botw from '../assets/botw.jpeg';
import { useParams, Link } from 'react-router-dom';
import { reviews, Data, gameData } from '@/components/GameData';
import { Button } from '@/components/ui/button';
import { ArrowLeft } from 'lucide-react';
import Rating from '@/components/Rating';
import {
  Card,
  CardContent,
  CardFooter,
  CardHeader,
  CardTitle,
} from '@/components/ui/card';
import ReviewCard from '@/components/ReviewCard';
import ReviewModal from '@/components/ReviewModal';

type GameDetailParams = {
  id: string;
};

const BaseGameDetailPage = () => {
  const { id } = useParams<GameDetailParams>();
  const index = id ? parseInt(id, 10) : NaN;
  const game: Data = gameData[index];

  // Check if the game data exists
  if (!game) {
    return <div>Game not found</div>;
  }

  return (
    <div className="flex justify-center">
      {/* Back to home button*/}
      <div className="grid max-w-[1500px] grid-cols-1 gap-4 px-4">
        <Link to="/" className="w-10" aria-label="back to home">
          <Button
            size="icon"
            variant="ghost"
            className="rounded-2x flex-shrink-0"
          >
            <ArrowLeft />
          </Button>
        </Link>
        <div className="grid gap-2 lg:grid-cols-[auto,1fr]">
          {/* Image, ratings, Write Review */}
          <Card className="overflow-hidden p-0">
            <CardHeader className="p-0">
              <div className="flex w-full cursor-default p-0">
                <img
                  src={game.image || Botw}
                  alt={game.title}
                  className="h-full max-h-[300px] w-full object-cover"
                  loading="lazy"
                />
              </div>
              <div className="mt-2 flex items-center justify-center text-yellow-400">
                <Rating rating={game.rating} numRatings={41} />
              </div>
            </CardHeader>
            <CardFooter className="flex flex-col justify-center">
              <ReviewModal />
            </CardFooter>
          </Card>

          {/* Title, Release Date, Platforms, Genres and Description */}
          <Card className="pb-4 text-left lg:w-[600px]">
            <CardHeader className="flex flex-col items-start">
              <CardTitle className=" text-4xl font-semibold">
                {game.title}
              </CardTitle>
              <CardContent className="py-2">
                <div className="flex flex-col justify-start">
                  <div className="flex">
                    <p>Release Date: 12.12.2017 </p>
                  </div>
                  <div className="mt-1 flex flex-row flex-wrap">
                    <p className="mr-2">Platforms:</p>
                    {game.platforms?.map(platform => (
                      <li
                        className="mr-1 list-none rounded-lg border border-primary px-2 text-sm"
                        key={platform}
                      >
                        {platform}
                      </li>
                    ))}
                  </div>
                  <div className="mt-1 flex flex-row flex-wrap">
                    <p className="mr-2">Genres:</p>
                    {game.genres?.map(genre => (
                      <li
                        className="mr-1 list-none rounded-lg border border-primary px-2 text-sm"
                        key={genre}
                      >
                        {genre}
                      </li>
                    ))}
                  </div>
                </div>
              </CardContent>
            </CardHeader>
            <CardContent className="py-4">
              <p className="text-md text-muted-foreground">
                {game.description}
              </p>
            </CardContent>
          </Card>
          {/* Reviews */}
          <div className="col-span-1 flex w-full flex-col justify-center text-left lg:w-[500px]">
            <h1 className="text-2xl font-bold text-foreground">Reviews</h1>
            {reviews.map(data => (
              <div key={data.id} className="my-1">
                <ReviewCard review={data} />
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};

const GameDetailPage = withLayout(BaseGameDetailPage);
export default GameDetailPage;
