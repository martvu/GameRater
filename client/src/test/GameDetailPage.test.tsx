import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import GameDetailPage from '@/pages/GameDetailPage';
import { vi, describe, it } from 'vitest';
import { allMocks as mocks } from '../mocks/mockQueries';
import { testPageRender } from './test-utils';
import userEvent from '@testing-library/user-event';

beforeAll(() => {
  // Mock `window.scrollTo` to accept any arguments
  window.scrollTo = vi.fn((x?: number | ScrollToOptions, y?: number) => {});
});

// Mock 'react-router-dom'
vi.mock('react-router-dom', async () => {
  const actualModule =
    await vi.importActual<typeof import('react-router-dom')>(
      'react-router-dom'
    );

  return {
    ...actualModule,
    useParams: vi.fn(() => ({ id: '1' })),
  };
});

describe('GameDetailPage Component', () => {
  it('renders the GamesList component with mocked GraphQL data', async () => {
    testPageRender(<GameDetailPage />, { mocks });
    // Wait for the mocked data to be displayed
    await waitFor(() => {
      expect(screen.getByText('Example Game')).toBeInTheDocument();
      expect(screen.getByText('Reviews')).toBeInTheDocument();
      expect(screen.getByText('Add Review')).toBeInTheDocument();
      expect(screen.getByText('Based on 2 user ratings')).toBeInTheDocument();
      expect(screen.getByText('4.5/5')).toBeInTheDocument();
    });
  });

  it('prompts sign in when trying to add a review', async () => {
    testPageRender(<GameDetailPage />, { mocks });
    const addReviewButton = await screen.findByText('Add Review');
    userEvent.click(addReviewButton);

    await waitFor(() => {
      expect(screen.getByText('Please sign in to review this game!')).toBeInTheDocument();
    });
  });
});
