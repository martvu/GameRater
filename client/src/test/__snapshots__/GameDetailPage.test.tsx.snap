// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameDetailPage Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-screen flex-col"
  >
    <header
      class="fixed top-0 z-40 flex h-16 w-[calc(100vw-0.75rem)] items-center bg-background/95 backdrop-blur supports-[backdrop-filter]:bg-background/80"
    >
      <nav
        class="flex w-full items-center justify-between"
      >
        <a
          aria-label="Return to Home Page"
          data-testid="logo-btn"
          href="/"
        >
          <div
            class="w-28 items-center justify-between gap-2 px-2 flex"
          >
            <button
              class="inline-flex items-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 justify-start hover:text-gray-400 h-10 px-4 py-2 hover:opacity-80"
            >
              <span
                class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
              >
                <span
                  class="flex h-full w-full items-center justify-center rounded-full bg-muted"
                >
                  CN
                </span>
              </span>
              <h1
                class="ml-1 text-xl font-bold tracking-wide text-primary"
              >
                GameRater
              </h1>
            </button>
          </div>
        </a>
        <div
          class="w-full max-w-sm flex-grow items-center justify-center space-x-2 md:flex hidden md:flex"
        >
          <form
            class="flex"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
              for="search"
            />
            <div
              class="relative w-[200px] md:w-[300px] lg:w-[400px]"
            >
              <input
                class="flex h-10 border border-input px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 w-full rounded-xl bg-muted pr-24 shadow-inner md:w-full lg:w-full"
                data-testid="search-input"
                placeholder="Search"
                type="text"
                value=""
              />
              <button
                aria-label="Search"
                class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-transparent text-secondary-foreground hover:text-muted-foreground transition duration-300 ease-in-out h-10 py-2 absolute right-0 top-0 rounded-l-none rounded-r-xl px-3 active:opacity-90"
                type="submit"
              >
                <svg
                  class="lucide lucide-search"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <path
                    d="m21 21-4.3-4.3"
                  />
                </svg>
              </button>
            </div>
          </form>
        </div>
        <div
          class="flex-shrink-0 items-center md:gap-2 flex"
        >
          <button
            class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-2xl md:hidden"
          >
            <svg
              class="lucide lucide-search"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <path
                d="m21 21-4.3-4.3"
              />
            </svg>
          </button>
          <div>
            <button
              aria-controls="radix-:r0:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
              data-state="closed"
              data-testid="modal-sign-in-button"
              type="button"
            >
              Sign In
            </button>
          </div>
          <div
            class="mr-1"
          >
            <button
              class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-2xl"
            >
              <svg
                class="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <path
                  d="M12 2v2"
                />
                <path
                  d="M12 20v2"
                />
                <path
                  d="m4.93 4.93 1.41 1.41"
                />
                <path
                  d="m17.66 17.66 1.41 1.41"
                />
                <path
                  d="M2 12h2"
                />
                <path
                  d="M20 12h2"
                />
                <path
                  d="m6.34 17.66-1.41 1.41"
                />
                <path
                  d="m19.07 4.93-1.41 1.41"
                />
              </svg>
              <svg
                class="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Toggle theme
              </span>
            </button>
          </div>
        </div>
      </nav>
    </header>
    <div
      class="flex-grow"
    >
      <div
        class="flex justify-center pt-20"
      >
        <div
          class="grid max-w-[1200px] grid-cols-1 gap-4 px-4"
        >
          <button
            aria-label="Back to previous page"
            class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-2x flex-shrink-0"
          >
            <svg
              class="lucide lucide-arrow-left"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
          </button>
          <main
            class="grid gap-2 md:grid-cols-[auto,1fr]"
          >
            <section
              class="flex flex-col items-center justify-center gap-4"
            >
              <div
                class="rounded-lg border bg-card text-card-foreground shadow-lg w-auto overflow-hidden p-0"
              >
                <div
                  class="flex flex-col h-[364px] w-[264px] p-0"
                >
                  <div
                    class="relative flex cursor-default p-0"
                  >
                    <div
                      class="absolute right-2 top-2"
                    >
                      <button
                        aria-label="Favorite game button"
                        class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 w-10 rounded-full z-10"
                        data-testid="favorite-btn"
                      >
                        <svg
                          class="none"
                          data-testid="heart-icon"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                          />
                        </svg>
                      </button>
                    </div>
                    <img
                      alt="Example Game"
                      class="object-cover"
                      loading="lazy"
                      src="https://images.igdb.com/igdb/image/upload/t_cover_big/image1.jpg"
                    />
                  </div>
                </div>
                <div
                  class="pt-0"
                >
                  <div
                    class=" flex items-center justify-center text-yellow-400"
                  >
                    <div
                      class="flex flex-col items-center"
                    >
                      <div
                        class="flex items-center font-light"
                      >
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-muted-foreground"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ☆  
                          </span>
                        </button>
                        <span
                          class="ml-2 text-muted-foreground"
                        >
                          4.5/5
                        </span>
                      </div>
                      <span
                        class="text-xs leading-6 text-muted-foreground"
                      >
                        Based on 2 user ratings
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="items-center pt-0 flex flex-col justify-center"
                >
                  <button
                    aria-controls="radix-:r6:"
                    aria-expanded="false"
                    aria-haspopup="dialog"
                    class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 active:bg-primary/70 h-10 px-4 py-2 mb-4 w-[200px]"
                    data-state="closed"
                    data-testid="add-review-btn"
                    type="button"
                  >
                    Add Review
                  </button>
                </div>
              </div>
            </section>
            <div
              class="rounded-lg border p-4 text-card-foreground shadow-lg border-none bg-transparent pb-4 text-left md:min-w-[400px] md:max-w-[700px] lg:min-w-[500px]"
            >
              <div
                class="flex flex-col items-start"
              >
                <h3
                  class="tracking-tight text-4xl font-semibold"
                >
                  Example Game
                </h3>
                <div
                  class="py-2"
                >
                  <div
                    class="flex flex-col justify-start gap-1"
                  >
                    <div
                      class="flex items-center gap-2"
                    >
                      <p
                        class="text-muted-foreground"
                      >
                        Metascore: 
                      </p>
                      <div
                        aria-label="metascore"
                        class="flex items-center justify-center rounded-md border border-white text-sm text-white bg-green-600 h-7 w-7 px-1"
                      >
                        85
                      </div>
                    </div>
                    <div
                      class="mt-1 flex items-center gap-2"
                    >
                      <p
                        class="text-muted-foreground"
                      >
                        Release Date:
                      </p>
                      <p
                        class="text-sm font-semibold"
                      >
                        03.05.2021
                      </p>
                    </div>
                    <div
                      class="mt-1 flex flex-row flex-wrap"
                    >
                      <p
                        class="mr-2 text-muted-foreground"
                      >
                        Platforms:
                      </p>
                      <li
                        class="mb-1 mr-1 list-none"
                      >
                        <div
                          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80"
                        >
                          PC
                        </div>
                      </li>
                    </div>
                    <div
                      class="mt-1 flex flex-row flex-wrap"
                    >
                      <p
                        class="mr-2 text-muted-foreground"
                      >
                        Genres:
                      </p>
                      <li
                        class="mb-1 mr-1 list-none"
                      >
                        <div
                          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80"
                        >
                          Adventure
                        </div>
                      </li>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="py-4"
              >
                <p
                  class="text-md"
                >
                  This is an example game summary.
                </p>
              </div>
            </div>
            <div
              class="col-span-1 flex h-full w-full justify-center lg:col-span-2"
            >
              <div
                class="flex min-w-full flex-col justify-center text-left lg:min-w-[700px]"
              >
                <h1
                  class="text-2xl font-bold text-foreground"
                >
                  Reviews
                </h1>
                <div
                  class="my-2"
                >
                  <div
                    class="rounded-lg border bg-card p-4 text-card-foreground shadow-lg"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <div
                        class="flex flex-col"
                      >
                        <p
                          class="text-lg font-semibold"
                        >
                          Great Game
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          user1
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          PC
                        </p>
                      </div>
                      <div
                        class="flex items-center font-light"
                      >
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <span
                          class="ml-2 text-muted-foreground"
                        >
                          5/5
                        </span>
                      </div>
                    </div>
                    <p
                      class="text-sm text-muted-foreground flex max-w-[800px] flex-row items-center gap-2"
                    >
                      This is a great game because...
                    </p>
                  </div>
                </div>
                <div
                  class="my-2"
                >
                  <div
                    class="rounded-lg border bg-card p-4 text-card-foreground shadow-lg"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <div
                        class="flex flex-col"
                      >
                        <p
                          class="text-lg font-semibold"
                        >
                          Good Game
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          user2
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          PC
                        </p>
                      </div>
                      <div
                        class="flex items-center font-light"
                      >
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-red-400"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ★  
                          </span>
                        </button>
                        <button
                          class="text-muted-foreground"
                          data-testid="review-star"
                          disabled=""
                          type="button"
                        >
                          <span
                            class="text-2xl font-light"
                          >
                            ☆  
                          </span>
                        </button>
                        <span
                          class="ml-2 text-muted-foreground"
                        >
                          4/5
                        </span>
                      </div>
                    </div>
                    <p
                      class="text-sm text-muted-foreground flex max-w-[800px] flex-row items-center gap-2"
                    >
                      This is a good game because...
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </main>
        </div>
      </div>
    </div>
    <footer
      class="mt-5 h-16 w-full border-t text-slate-400"
    >
      <div
        class="container mx-auto px-6 pt-5 text-left text-sm"
      >
        Copyright 2023 GameRater
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
